lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -353,8 +358,21 @@ static EnumDecl *synthesizeCodingKeysEnum(DerivedConformance &derived) {
       }
 
       case DoesNotConform:
+        // for accurate diagnostic, we show TypeRepr if present for IUO
+        TypeLoc typeLoc = TypeLoc();
+        if(varDecl->isImplicitlyUnwrappedOptional()){
+            typeLoc = {
+                varDecl->getTypeReprOrParentPatternTypeRepr(),
+                varDecl->getType(),
+            };
+        } else {
+            typeLoc = {
+                NULL,
```

Comment:
You don't need to conditionalize this. `getTypeReprOrParentPatternTypeRepr` will always get what the user wrote, if they wrote something at all.

Reply:
@CodaFi 

Yes, but this is breaking nested class/struct types. Please see the test regression I have commented in the https://bugs.swift.org/browse/SR-13119. @varungandhi-apple suggested a change which I inferred to be the above.

Reply:
Name lookup will see the innermost name in that scope and it should just work out. As long as those fixits compile, this is not a regression and we should update those tests. I very much prefer we preserve the way the user spelled types than fully qualify them all the time.

Reply:
Sure, okay. I get it

Reply:
> I very much prefer we preserve the way the user spelled types than fully qualify them all the time.

I didn't realize there was a disagreement here. Should we write this down somewhere so that newer diagnostics use this style? I thought this was a regression because the original code was printing the full type.

Reply:
That's a good point. I recall we have a document that tries to explain good diagnostic hygiene. And this is certainly one of those places where you have to be really careful. Because the user-provided type may not always work if you're pulling something from a different DeclContext, and the fully-qualified type may not even work when you have ambiguities caused by e.g. the module-and-type-sharing-names bug.